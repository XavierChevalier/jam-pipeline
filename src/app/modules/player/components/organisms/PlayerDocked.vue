<script setup lang="ts">
  import { PropType } from 'vue'
  import { Track } from '@/app/modules/track/models/Track'
  import DotSeparator from '@/app/components/atoms/AppDotSeparator.vue'
  import PlayerProgressBar from '@/app/modules/player/components/atoms/PlayerProgressBar.vue'
  import IconLike from '@/app/components/molecules/AppIconLike.vue'
  import Icon from '@/app/components/atoms/AppIcon.vue'
  import { mdiPlayOutline } from '@mdi/js'
  import TrackCover from '@/app/modules/track/components/atoms/TrackCover.vue'
  import { isPropertyValid } from '@/app/tools/component-properties/PropertyValidator'
  import PlayerRemoteListeningDevice from '@/app/modules/player/components/molecules/PlayerRemoteListeningDevice.vue'

  defineProps({
    track: {
      type: Object as PropType<Track>,
      required: true,
      validator: isPropertyValid(Track),
    },
    remoteListeningDevice: {
      type: String,
      required: false,
      default: null,
    },
  })
</script>

<template>
  <div>
    <div
      class="
        text-xs
        flex
        px-5
        py-3
        rounded
        shadow-lg
        bg-primary-light
        items-center
      "
    >
      <TrackCover class="mr-3" size="sm" :track="track" />

      <div class="flex flex-col">
        <div>
          <span class="font-bold">{{ track.title }}</span>
          <DotSeparator />
          <span v-for="band in track.bands" :key="band">{{ band }}</span>
        </div>

        <PlayerRemoteListeningDevice
          v-if="remoteListeningDevice"
          :remote-device-name="remoteListeningDevice"
        />
      </div>

      <div class="flex ml-auto">
        <IconLike class="mr-3 text-black" />
        <Icon :path="mdiPlayOutline" />
      </div>
    </div>

    <PlayerProgressBar class="mx-5 translate-y-[-50%]" :progress="50" />
  </div>
</template>
