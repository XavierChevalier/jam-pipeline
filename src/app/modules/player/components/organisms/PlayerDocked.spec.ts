import * as Stories from '@/app/modules/player/components/organisms/PlayerDocked.stories'
import { StorybookTestsGenerator } from '@/tests/tests-generators/StorybookTestsGenerator'

describe('PlayerDocked', () => {
  StorybookTestsGenerator.fromStoriesExports(Stories)
    .snapshotEachStories()
    .property<string>('remoteListeningDevice', (property) => {
      property
        .itShouldBeDefined()
        .itShouldNotBeRequired()
        .itShouldHaveADefaultValue()
    })
})
