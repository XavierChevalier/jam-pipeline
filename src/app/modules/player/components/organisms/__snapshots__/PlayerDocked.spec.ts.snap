// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerDocked stories snapshots [snapshot] should render the "Default" component story correctly 1`] = `
<div>
  <div
    class="text-xs flex px-5 py-3 rounded shadow-lg bg-primary-light items-center"
  >
    <anonymous-stub
      class="mr-3"
      size="sm"
      track="[object Object]"
    />
    <div
      class="flex flex-col"
    >
      <div>
        <span
          class="font-bold"
        >
          ANSI.SYS
        </span>
        <anonymous-stub />
        
        <span>
          MASTER BOOT RECORD
        </span>
        
      </div>
      <!--v-if-->
    </div>
    <div
      class="flex ml-auto"
    >
      <anonymous-stub
        class="mr-3 text-black"
        counter="0"
        liked="false"
        showcounter="false"
      />
      <anonymous-stub
        path="M8.5,8.64L13.77,12L8.5,15.36V8.64M6.5,5V19L17.5,12"
      />
    </div>
  </div>
  <anonymous-stub
    class="mx-5 translate-y-[-50%]"
    progress="50"
  />
</div>
`;

exports[`PlayerDocked stories snapshots [snapshot] should render the "ListeningOnDesktop" component story correctly 1`] = `
<div>
  <div
    class="text-xs flex px-5 py-3 rounded shadow-lg bg-primary-light items-center"
  >
    <anonymous-stub
      class="mr-3"
      size="sm"
      track="[object Object]"
    />
    <div
      class="flex flex-col"
    >
      <div>
        <span
          class="font-bold"
        >
          ANSI.SYS
        </span>
        <anonymous-stub />
        
        <span>
          MASTER BOOT RECORD
        </span>
        
      </div>
      <anonymous-stub
        remotedevicename="Desktop-B9M8G7J"
      />
    </div>
    <div
      class="flex ml-auto"
    >
      <anonymous-stub
        class="mr-3 text-black"
        counter="0"
        liked="false"
        showcounter="false"
      />
      <anonymous-stub
        path="M8.5,8.64L13.77,12L8.5,15.36V8.64M6.5,5V19L17.5,12"
      />
    </div>
  </div>
  <anonymous-stub
    class="mx-5 translate-y-[-50%]"
    progress="50"
  />
</div>
`;
